onadata/libs/filters.py

```diff
@@ -399,6 +401,24 @@ def filter_queryset(self, request, queryset, view):
         return queryset
 
 
+class AttachmentTypeFilter(filters.BaseFilterBackend):
+
+    def filter_queryset(self, request, queryset, view):
+        attachment_type = request.query_params.get('type')
+
+        # return instances with image specific attachments.
+        if attachment_type == "image":
+            queryset = queryset.filter(mimetype__in=MEDIA_IMAGE_TYPES)
+        elif attachment_type == "audio":
+            queryset = queryset.filter(mimetype__in=MEDIA_AUDIO_TYPES)
+        elif attachment_type == "video":
+            queryset = queryset.filter(mimetype__in=MEDIA_VIDEO_TYPES)
+        elif attachment_type == "file":
+            queryset = queryset.filter(mimetype__in=MEDIA_FILE_TYPES)
+
```

Comment:
Feels like too many if statements, we could refactor to use a dictionary for all this types, i.e.
```python
MEDIA_FILE_TYPES = {
  "audio": ["audio/mp3", "audio/x-m4a", ".."],
    "documents": ["application/msword", "application/vnd.ms-excel", "..."],
    "image": ["image/jpeg", "image/png", "..."],
    ...
}
...
mime_types = MEDIA_FILE_TYPES.get(attachment_type)

if mime_types:
   queryset = queryset.filter(mimetype__in=mimetypes)
````